p2p/src/sync/tests/tx_announcement.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use chainstate::ban_score::BanScore;
+use common::{
+    chain::{
+        config::create_unit_test_config, signature::inputsig::InputWitness, tokens::OutputValue,
+        OutPointSourceId, OutputPurpose, SignedTransaction, Transaction, TxInput, TxOutput,
+    },
+    primitives::Amount,
+};
+use mempool::error::{Error as MempoolError, TxValidationError};
+
+use crate::{
+    message::Announcement, sync::tests::helpers::SyncManagerHandle, types::peer_id::PeerId,
+    P2pError,
+};
+
+#[tokio::test]
+#[should_panic = "Received a message from unknown peer"]
+async fn nonexistent_peer() {
+    let mut handle = SyncManagerHandle::builder().build().await;
+
+    let peer = PeerId::new();
+
+    let tx = Transaction::new(0x00, vec![], vec![], 0x01).unwrap();
+    let tx = SignedTransaction::new(tx.clone(), vec![]).unwrap();
```

Comment:
No need to clone here or in any of the tests below.

I don't know why clippy isn't catching this.